File: ruby/fib.rb
def fibonacci(n)
  return n if n <= 1
  return fibonacci(n - 1) + fibonacci(n - 2)
end

begin
  sequences = ARGV[0].to_i
  puts "Ruby Fib: #{sequences}x"
  start = Process.clock_gettime(Process::CLOCK_MONOTONIC)
  output = fibonacci(sequences)
  finish = Process.clock_gettime(Process::CLOCK_MONOTONIC)
  elapsed = (finish - start) * 1000
  puts "Result: #{output} | Time: #{elapsed.to_i}ms"
rescue Exception => e
  puts e
end

File: elixir/fib.exs
defmodule Fib do
  def fibonacci(n) when n <= 1 do
    n
  end

  def fibonacci(n) do
    fibonacci(n - 1) + fibonacci(n - 2)
  end
end

try do
  [arg1] = System.argv
  sequences = String.to_integer(arg1)
  IO.puts "Elixir Fib: #{sequences}x"
  start = :os.system_time(:millisecond)
  output = Fib.fibonacci(sequences)
  finish = :os.system_time(:millisecond)
  elapsed = (finish - start)
  IO.puts "Result: #{output} | Time: #{elapsed}ms"
rescue
  e -> IO.puts e
end
